addons/website_slides/models/slide_channel.py

```diff
@@ -38,21 +38,14 @@ def _recompute_completion(self):
             mapped_data.setdefault(item['channel_id'][0], dict())
             mapped_data[item['channel_id'][0]][item['partner_id'][0]] = item['__count']
 
-        partner_karma = dict.fromkeys(self.mapped('partner_id').ids, 0)
         for record in self:
             record.completed_slides_count = mapped_data.get(record.channel_id.id, dict()).get(record.partner_id.id, 0)
             record.completion = 100.0 if record.completed else round(100.0 * record.completed_slides_count / (record.channel_id.total_slides or 1))
             if not record.completed and record.completed_slides_count >= record.channel_id.total_slides:
                 record.completed = True
-                partner_karma[record.partner_id.id] += record.channel_id.karma_gen_channel_finish
-
-        partner_karma = {partner_id: karma_to_add
-                         for partner_id, karma_to_add in partner_karma.items() if karma_to_add > 0}
-
-        if partner_karma:
-            users = self.env['res.users'].sudo().search([('partner_id', 'in', list(partner_karma.keys()))])
-            for user in users:
-                users.add_karma(partner_karma[user.partner_id.id])
+                user = self.env['res.users'].sudo().search([('partner_id', '=', record.partner_id.id)])[0]
```

Comment:
search in loop ! 
Is it not possible to make a list of ids lie it was done before, to search only once and loop through the search result ?

Reply:
Ok, done! As I am not yet a Python expert do not hesitate to make any comment on the modification I make :)